### PR TITLE
Avoid linking AO aux tables to extension

### PR DESCRIPTION
**Pull request by InnerLife0**
Avoid linking AO aux tables to extension

Unlike TOAST tables, AO aux tables are linked to extension during its creation.
When table is dropped from extension using ALTER, aux tables are left as
extension dependencies. It means we can't DROP a table after that, because AO
aux tables are still used by extension. In other words, this behavior makes
impossible to DROP AO table that was created by extension without extension
DROP.

The goal of the patch is to change this behavior, allowing to DROP a table.
Unnecessary dependencies are now not created in heap_create_with_catalog(). The
behavior is the same as for TOAST tables, which are aux tables too. The
dependency created in CreateAOAuxiliaryTable() is enough.

The patch contains a test that describes target behavior.

This is a backport of dfe49e33d2c8ef04db1e511cb979fe1bb86bb805. The test was
changed as 6x has no test_ext* extensions. To achieve the same result, new test
extension was made.

### CODE REVIEW

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/68638

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1259716

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1259717

**Review by RekGRpth — DISMISSED** [review]

**Review by bandetto — DISMISSED** [review]

File: src/test/modules/test_extensions/sql/test_extensions.sql [hunk 1]
```diff
@@ -185,3 +185,22 @@ show search_path;
...
 drop schema issue6716 cascade;
+
+--
+-- Test that extension not depends on AO aux tables
+--
+
+create extension test_ext_ao;
+-- Detach tables from extension.
+alter extension test_ext_ao drop table test_ext_ao_table;
+-- The following select should give 0 rows, as there no extension related
```
Comment (andr-sokolov) [inline src/test/modules/test_extensions/sql/test_extensions.sql]: may be "there **are** no extension related tables left **there**"?

Comment (RekGRpth) [reply src/test/modules/test_extensions/sql/test_extensions.sql]: original commit has exact same sentence

Comment (InnerLife0) [reply src/test/modules/test_extensions/sql/test_extensions.sql]: Fixed. The last word is unnecessary here.

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/68803

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1271570

File: src/test/modules/test_extensions/expected/test_extensions.out [hunk 2]
```diff
@@ -276,3 +276,25 @@ drop schema issue6716 cascade;
 NOTICE:  drop cascades to 2 other objects
 DETAIL:  drop cascades to function issue6716.test_func1(integer,integer)
 drop cascades to extension test_ext_cau
+--
+-- Test that extension not depends on AO aux tables
```
Comment (bandetto) [inline src/test/modules/test_extensions/expected/test_extensions.out:280]: Perhaps this wording is better?
```suggestion
-- Test that extension does not depend on AO aux tables
```
That sentence is in some other places too.

Comment (InnerLife0) [reply src/test/modules/test_extensions/expected/test_extensions.out:280]: I don't think it matters.

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1271571

**Review by bandetto — APPROVED** [review]

**Review by RekGRpth — APPROVED** [review]

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/69079

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1284045

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1284047

**Comment by BenderArenadata** [comment]
Failed job Behave tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1284043

**Pull request merged**